### ISSUE TITLE
Add fixed geometries in catchments shp `p1_catchments_shp` to pipeline

### ISSUE THREAD

**Issue by msleckman** [USGS-R/drb-inland-salinity-ml#60]
Fix Geometries in catchments shp p1_catchments_shp in pipeline. 

Based on discussion in #54

**Comment by lekoenig** [comment]
Does this work? 

```
  # Create zero-width buffer around HRU polygons to make geometries valid
  p1_catchments_sf_valid <- st_buffer(p1_catchments_sf,0)
```

**Comment by jds485** [comment]
From discussion in #45, we should update the original file of PRMS HRU areas to instead use the NHD catchment areas for all reaches x that have x_n and n > 1. For our data, those are 3_1, 3_2, 8_1, 8_2, 12_1, and 12_2.

**Comment by msleckman** [comment]
I added this target in #38 
https://github.com/USGS-R/drb-inland-salinity-ml/pull/38#discussion_r794101443

which fixes geometry issues that have been arising when running spatial processing. This issue will close once #38 is merged.

> From discussion in #45, we should update the original file of PRMS HRU areas to instead use the NHD catchment areas for all reaches x that have x_n and n > 1. For our data, those are 3_1, 3_2, 8_1, 8_2, 12_1, and 12_2.
This is resolved now via #45, correct?

**Comment by jds485** [comment]
> From discussion in #45, we should update the original file of PRMS HRU areas to instead use the NHD catchment areas for all reaches x that have x_n and n > 1. For our data, those are 3_1, 3_2, 8_1, 8_2, 12_1, and 12_2.

I don't think that's resolved yet. This would add a `p2_` target that edits the catchment areas for these reaches to match the NHD areas. Do you want to take care of that and add to #38?

**Comment by msleckman** [comment]
Would that be a large addition? I can't tell without looking further into it. 
But #38 has blocking a few other PRs, so i'd suggest merging #38 and working on this issue in separate PR.

**Comment by jds485** [comment]
If #38 is ready to be merged now, this can be addressed in a new PR. I don't expect that it would be a large addition. I think it would affect the resulting values assigned to the reaches in #38 because the areas are updated.

**Comment by lekoenig** [comment]
Here are the tasks to be completed to resolve this issue:

- [ ] Make HRU polygons valid by adding a zero-width buffer around HRU polygons

- [ ] Use NHDv2 catchments for PRMS segments that were split within the temperature pipeline

> update the original file of PRMS HRU areas to instead use the NHD catchment areas for all reaches x that have x_n and n > 1. For our data, those are 3_1, 3_2, 8_1, 8_2, 12_1, and 12_2.

- [ ] For PRMS segments that have no corresponding HRUs, use dissolved NHDv2 catchments

The polygons resulting from these steps will represent a new target in the process phase of the pipeline.

**Comment by jds485** [comment]
I changed the name of this issue because we can now add the fixed geometries from the `drb-network-prep` repo to our repo.
